File: android/src/test/java/com/liferay/mobile/sdk/test/portal/OrderByComparatorTest.java
/**
 * Copyright (c) 2000-2014 Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.sdk.test.portal;

import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.v62.bookmarksentry.BookmarksEntryService;
import com.liferay.mobile.sdk.test.BaseTest;

import java.io.IOException;

import org.json.JSONArray;
import org.json.JSONObject;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author Bruno Farache
 */
public class OrderByComparatorTest extends BaseTest {

	public OrderByComparatorTest() throws IOException {
		super();
	}

	@Test
	public void getEntriesDescending() throws Exception {
		BookmarksEntryService service = new BookmarksEntryService(session);
		long groupId = props.getGroupId();

		String className =
			"com.liferay.portlet.bookmarks.util.comparator.EntryNameComparator";

		JSONObjectWrapper orderByComparatorDesc = new JSONObjectWrapper(
			className, new JSONObject());

		JSONArray entries = service.getEntries(
			groupId, ServiceContextTest.PARENT_FOLDER_ID, -1, -1,
			orderByComparatorDesc);

		assertEquals(2, entries.length());

		JSONObject first = entries.getJSONObject(0);
		JSONObject second = entries.getJSONObject(1);

		assertEquals("Z", first.getString("name"));
		assertEquals("A", second.getString("name"));
	}

	@Before
	public void setUp() throws Exception {
		_serviceContextTest = new ServiceContextTest();
		_entryA = _serviceContextTest.addEntry("A", null);
		_entryZ = _serviceContextTest.addEntry("Z", null);
	}

	@After
	public void tearDown() throws Exception {
		_serviceContextTest.deleteEntry(_entryA);
		_serviceContextTest.deleteEntry(_entryZ);
	}

	private JSONObject _entryA;
	private JSONObject _entryZ;
	private ServiceContextTest _serviceContextTest;

}

File: windows/Liferay.SDK.Test/OrderByComparatorTest.cs
//------------------------------------------------------------------------------
// <copyright file="OrderByComparatorTest.cs">
//    Copyright (c) 2014-present Andrea Di Giorgi
//
//    Permission is hereby granted, free of charge, to any person obtaining a
//    copy of this software and associated documentation files (the "Software"),
//    to deal in the Software without restriction, including without limitation
//    the rights to use, copy, modify, merge, publish, distribute, sublicense,
//    and/or sell copies of the Software, and to permit persons to whom the
//    Software is furnished to do so, subject to the following conditions:
//
//    The above copyright notice and this permission notice shall be included in
//    all copies or substantial portions of the Software.
//
//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
//    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
//    DEALINGS IN THE SOFTWARE.
// </copyright>
// <author>Andrea Di Giorgi</author>
// <website>https://github.com/Ithildir/liferay-sdk-builder-windows</website>
//------------------------------------------------------------------------------

using System.Linq;
using System.Threading.Tasks;
using Liferay.SDK.Service.V62.BookmarksEntry;
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;

namespace Liferay.SDK.Test
{
	[TestClass]
	public class OrderByComparatorTest : TestBase
	{
		private dynamic entryA;
		private dynamic entryZ;
		private ServiceContextTest serviceContextTest;

		[TestCleanup]
		public async Task Cleanup()
		{
			await this.serviceContextTest.DeleteBookmarkEntryAsync(this.entryA);
			await this.serviceContextTest.DeleteBookmarkEntryAsync(this.entryZ);
		}

		[TestInitialize]
		public async Task Initialize()
		{
			this.serviceContextTest = new ServiceContextTest();

			this.entryA = await this.serviceContextTest.AddBookmarkEntryAsync("A", null);
			this.entryZ = await this.serviceContextTest.AddBookmarkEntryAsync("Z", null);
		}

		[TestMethod]
		public async Task TestGetEntriesDescending()
		{
			var service = new BookmarksEntryService(this.Session);

			var className = "com.liferay.portlet.bookmarks.util.comparator.EntryNameComparator";

			var orderByComparator = new JsonObjectWrapper(className);

			var entries = await service.GetEntriesAsync(OrderByComparatorTest.GroupId, ServiceContextTest.ParentFolderId, -1, -1, orderByComparator);

			var first = entries.First();
			var second = entries.ElementAt(1);

			Assert.AreEqual("Z", first.name);
			Assert.AreEqual("A", second.name);
		}

		[TestMethod]
		public async Task TestNullOrderByComparator()
		{
			var service = new BookmarksEntryService(this.Session);

			var entries = await service.GetEntriesAsync(OrderByComparatorTest.GroupId, ServiceContextTest.ParentFolderId, -1, -1, null);

			Assert.AreEqual(2, entries.Count());

			var first = entries.First();
			var second = entries.ElementAt(1);

			Assert.AreEqual("A", first.name);
			Assert.AreEqual("Z", second.name);
		}
	}
}
